wp-content/themes/pub/wporg-learn-2020/functions.php

```diff
@@ -290,6 +290,42 @@ function wporg_archive_modify_query( WP_Query $query ) {
 }
 add_action( 'pre_get_posts', 'wporg_archive_modify_query' );
 
+/**
+ * Modify the workshop post type archive query to prioritize workshops in the user's locale.
+ *
+ * @param array    $clauses
+ * @param WP_Query $query
+ *
+ * @return array
+ */
+function wporg_archive_query_prioritize_locale( $clauses, $query ) {
+	if ( ! $query->is_post_type_archive( 'wporg_workshop' ) || is_admin() ) {
+		return $clauses;
+	}
+
+	global $wpdb;
+
+	$locale      = get_locale() ?: 'en_US';
+	$locale_root = preg_replace( '#^([a-z]{2,3}_?)[a-zA-Z_-]*#', '$1', $locale, -1, $count );
```

Comment:
If that doesn't work, then it might be worth opening an issue in the GP repo, and suggest that they add a property for the locale root. I think you're right that using the root will make more sense in a lot of contexts where the locale switcher will be used in the future, so that info could come in handy.

Reply:
I checked `lang_code_iso_639_1` (and also `lang_code_iso_639_2`) and unfortunately neither matches `wp_locale` in all cases. So for now, this regex will have to suffice. I suspect adding a `wp_locale_root` property to every locale in GP wouldn't get much traction, but if we start needing this value a lot, I could see having a `get_locale_root()` function that makes this easy.

Reply:
👍🏻